From 854f7973ba2ca50123b64ddc60adac04c61bde43 Mon Sep 17 00:00:00 2001
From: Raphael Hertzog <hertzog@debian.org>
Date: Thu, 2 Aug 2007 19:18:12 +0200
Subject: [PATCH] dpkg-source: exclude directories created by distributed VCS from a native tarball build

It's much more convenient to be able to start a build from the VCS tree
instead of having to make an export or similar. This is particularly true
with distributed VCS which tend to keep all their stuff in a single directory
at the root of the tree. Current list includes bzr/git/hg/darcs/arch.
---
 scripts/dpkg-source.pl |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/scripts/dpkg-source.pl b/scripts/dpkg-source.pl
index 1ed1213..20c98fb 100755
--- a/scripts/dpkg-source.pl
+++ b/scripts/dpkg-source.pl
@@ -56,7 +56,8 @@ my %type;		 # used by checktype
 my %filepatched;	 # used by checkdiff
 my %dirtocreate;	 # used by checkdiff
 
-my @tar_ignore;
+my @tar_ignore = qw(--exclude=.git --exclude=.bzr --exclude=_darcs
+		    --exclude=.hg --exclude={arch});
 
 use POSIX;
 use Fcntl qw (:mode);
-- 
1.5.2.4

